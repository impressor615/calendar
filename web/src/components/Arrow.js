import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';


const Arrow = ({ direction, onClick, ...rest }) => (
  <button
    {...rest}
    onClick={onClick}
    className={classnames(
      'arrow',
      rest.className,
      { [direction]: direction },
    )}
  />
);

Arrow.defaultProps = {
  direction: 'previous',
};

Arrow.propTypes = {
  direction: PropTypes.oneOf(['next', 'previous']),
  onClick: PropTypes.func.isRequired,
};

export default Arrow;