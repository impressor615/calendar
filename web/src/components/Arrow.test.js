import React from 'react';

import Arrow from 'components/Arrow';


describe('<Arrow />', () => {
  const onClick = jest.fn();
  const props = {
    direction: 'previous',
    onClick,
  };

  it('should be rendered properly', () => {
    const wrapper = shallow(<Arrow {...props} />);
    expect(wrapper.find('button.arrow.previous')).toHaveLength(1);

    wrapper.setProps({
      direction: 'next',
    });
    expect(wrapper.find('button.arrow.next')).toHaveLength(1);
  });

  it('onClick shoule be called when clicking arrow button', () => {
    const wrapper = shallow(<Arrow {...props} />);
    wrapper.find('button').simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);
  })

  it('should be matched with the snapshot', () => {
    const wrapper = shallow(<Arrow {...props} />);
    expect(wrapper).toMatchSnapshot();
  })
});
