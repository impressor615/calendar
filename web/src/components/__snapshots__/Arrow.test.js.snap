// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Arrow /> should be matched with the snapshot 1`] = `
<button
  className="arrow previous"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
/>
`;
